scopes/dependencies/dependency-resolver/dependency-resolver.docs.md

```diff
@@ -1,235 +1,389 @@
+    "proxy": "http://domain-one.proxy.com:8080",
+    "httpsProxy": "http://domain-two.proxy.com:8080"
   }
 }
 ```
 
-## Documentation
-
-### saved data
-The dependency-resolver extension will store in its data (part of the component model), the final dependencies after all the calculations, and also the dependencies calculated by the workspace configuration (and require detection) - without putting the component config rules and dependencies added by extensions (via the hooks)
-(see calculation flow below)
-
-### calculation flow
-The dependency calculation will be triggered by external player (probably the workspace / scope extensions)
-This player will pass to the dependency resolver
-* "raw component" (exact type tbd), this will have in general the component files, the component config, and the component extensions at least.
-* workspace config for the resolver (optional) - should be discussed, see the open question below
-#### dependencies by code analysis (import / require detection)
-The dependency resolver will start by analyzing the imports / require statements of the source code and intersect them with the dependencies from the workspace configuration -
-If it detects for example a `require('lodash');` it will search for lodash in the config. (if not found it will throw an error that the dependency is missing so it can't resolve its version)
-(This part must be extendable by third part extensions, exact syntax TBD - see open question)
-#### add the extensions themselves as dependencies
-If a component has extension defined for it, this extension should be considered as a dev dependency.
-#### dependencies added by extensions
-An extension might provide more dependencies / remove dependencies (uses the hook - see API below).
-The dependency resolver will take this dependencies configuration added by all the hook subscribers and merge them with themselves (2 extensions for the same component might add the same dependency in a different version) - exact strategy to be defined - see open question
-Then it will merge the final result with the result of the previous steps (dependencies configured here are stronger than those from the code analysis)
-#### dependencies added by component config
-The dependency resolver will read special rules configured in the component config and will merge them into the previous results.
-Since these rules added explicitly by the user, they are the strongest ones.
-
-### open issues
-* names of the workspace configuration fields
-* exact policy structure
-* hooks syntax
-* type that the calculation gets as input (the "raw component")
-* how we extend the import / require analysis to enable 3rd party extension to extend it?
-* What is the strategy to merge dependencies added by hooks between themselves (random / load order / alphabetically?)
-* More methods that should be exposed via the API
-* FileDependencies hook name
+##### A proxy can also be set in Bit's global configuration
```

Comment:
what about the proxy config from npmrc?
and the order (dep resolver config, npmrc, global config)